### ISSUE TITLE
v1.13.0 erroneously reports non-zero exit code

### ISSUE THREAD

**Issue by FND** [prettier/prettier#4599]
reduced test case:

```
$ prettier *.js --debug-check --list-different
```

* with v1.12.1, this reports exit status 0 if the code is properly formatted
* with v1.13.0, this reports exit status 1 even though the code is properly formatted

----

original test case: https://gist.github.com/FND/8cdcc6c60bd5430b2204fe930acc3eec/revisions

with v1.12.1, the following works as expected:

1.  actively reformat source code:

    ```
    $ prettier --write *.js
    ```

2.  ensure source code is formatted properly (e.g. as part of the test suite, continuous integration etc.):

    ```
    $ prettier --write *.js --debug-check --list-different --no-write
    ```

    here `--no-write` overrides `--write`, which is useful because we don't have to modify #1's base command - particularly in `package.json`:

    ```json
    "scripts": {
        "test": "npm run format -- --debug-check --list-different --no-write",
        "format": "prettier --write '*.js'"
    }
    ```

however, with v1.13.0, that second command fails with exit code 1, despite the source code being properly formatted

**Comment by FND** [comment]
Wow, that was quick - thanks a lot, @ikatyang!

**Comment by duailibe** [comment]
This is fixed in 1.13.3.
